hal/timer/periph.go

```diff
@@ -60,6 +80,78 @@ func (p *Periph) DisableClock() {
 	mx.CLK_EN_CENT.Unlock()
 }
 
-func (p *Periph) ResetISR(ch int) {
-	p.CH[ch].EOI.Load()
+func (p *Periph) SetInterval(ch int, nanoseconds int64) {
+	clk := p.Bus().Clock()
+
+	step := 1e9 / clk
+	period := uint32(nanoseconds / step)
+
+	if period < 0 || period > 2147483647 {
+		panic("timer: period outside of 32bit range")
+	}
+
+	p.ch[ch].load_count.Store(period)
+}
+
+func (p *Periph) ClearIRQ(ch int) {
+	p.ch[ch].eoi.Load()
+}
```

Comment:
Shouldn't they be a Channel methods? You'll easily access them this way: `p.Channel(n).ClearIRQ()`

Reply:
Forgot to remove setinterval, and missed commit of ClearIRQ, it will clear all channels at once with the corresponding register